### DISCUSSION TITLE
Error "Unexpected token < in JSON at position 0" after moving the api folder to app/[locale]

### DISCUSSION THREAD

**Discussion by Yedidya10** [amannn/next-intl#476] [Q&A]
### Description

Hi.

I use version: 3.0.0-beta.10

I read somewhere in the docs that all the folders in the app folder should be moved to app/[locale]

The problem happens to me as soon as I move the api folder there. I get the following error:

> - error SyntaxError: Unexpected token < in JSON at position 0
>      at JSON.parse (<anonymous>)
>      at AsyncResource.runInAsyncScope (node:async_hooks:203:9)
> - error SyntaxError: Unexpected token < in JSON at position 0
>      at JSON.parse (<anonymous>)
>      at AsyncResource.runInAsyncScope (node:async_hooks:203:9)
> digest: "3150237731"

I have no idea how to get to the source of the issue.

### Mandatory reproduction URL (CodeSandbox or GitHub repository)

https://github.com/Yedidya10/sharesphere

### Reproduction description

Steps to reproduce:
1. Run npm run dev
2. Open http://localhost:3000
3. You can see the error in the browser or VS console

### Expected behaviour

that the code will run properly.

**Comment by amannn** [comment]
Why do you want to move the api folder there? The `[locale]` folder should be used for all content routes where you want to apply internationalization. I see that you're using NextAuth, moving its API routes somehwere else might have unexpected side effects too.

I'll move this to a discussion since it's more of a usage question.

**Comment by Yedidya10** [comment]
I understood.

The thing is, when I leave the api folder inside the app - the code runs, but I get an error in the VS console (below):

It doesn't stop the app from working (for now).


> Warning: React.createElement: type is invalid -- expected a string (for built-in components) or a class/function (for composite components) but got: undefined. You likely forgot to export your component from the file it's defined in, or you might have mixed up default and named imports.
> Warning: React.createElement: type is invalid -- expected a string (for built-in components) or a class/function (for composite components) but got: null.
> - error node_modules\next\dist\compiled\react-server-dom-webpack\cjs\react-server-dom-webpack-server.edge.development.js (1684:8) @ attemptResolveElement
> - error Error: Unsupported Server Component type: Null
>     at stringify (<anonymous>)
> null
> Error: Unsupported Server Component type: Null
>     at attemptResolveElement (webpack-internal:///(sc_server)/./node_modules/next/dist/compiled/react-server-dom-webpack/cjs/react-server-dom-webpack-server.edge.development.js:1421:19)
>     at resolveModelToJSON (webpack-internal:///(sc_server)/./node_modules/next/dist/compiled/react-server-dom-webpack/cjs/react-server-dom-webpack-server.edge.development.js:1627:41)
>     at Object.toJSON (webpack-internal:///(sc_server)/./node_modules/next/dist/compiled/react-server-dom-webpack/cjs/react-server-dom-webpack-server.edge.development.js:1169:28)
>     at stringify (<anonymous>)
>     at processModelChunk (webpack-internal:///(sc_server)/./node_modules/next/dist/compiled/react-server-dom-webpack/cjs/react-server-dom-webpack-server.edge.development.js:2116:24)
>     at retryTask (webpack-internal:///(sc_server)/./node_modules/next/dist/compiled/react-server-dom-webpack/cjs/react-server-dom-webpack-server.edge.development.js:1882:38)
>     at performWork (webpack-internal:///(sc_server)/./node_modules/next/dist/compiled/react-server-dom-webpack/cjs/react-server-dom-webpack-server.edge.development.js:1920:21)
>     at Timeout.eval [as _onTimeout] (webpack-internal:///(sc_server)/./node_modules/next/dist/compiled/react-server-dom-webpack/cjs/react-server-dom-webpack-server.edge.development.js:1429:28)
>     at listOnTimeout (node:internal/timers:569:17)
>     at process.processTimers (node:internal/timers:512:7) {
>   digest: '135229182'
> }
> - error node_modules\next\dist\compiled\react-server-dom-webpack\cjs\react-server-dom-webpack-server.edge.development.js (1684:8) @ attemptResolveElement
> - error Error: Unsupported Server Component type: Null
>     at attemptResolveElement (webpack-internal:///(sc_server)/./node_modules/next/dist/compiled/react-server-dom-webpack/cjs/react-server-dom-webpack-server.edge.development.js:1421:19)
>     at resolveModelToJSON (webpack-internal:///(sc_server)/./node_modules/next/dist/compiled/react-server-dom-webpack/cjs/react-server-dom-webpack-server.edge.development.js:1627:41)
>     at Object.toJSON (webpack-internal:///(sc_server)/./node_modules/next/dist/compiled/react-server-dom-webpack/cjs/react-server-dom-webpack-server.edge.development.js:1169:28)
>     at stringify (<anonymous>)
>     at processModelChunk (webpack-internal:///(sc_server)/./node_modules/next/dist/compiled/react-server-dom-webpack/cjs/react-server-dom-webpack-server.edge.development.js:2116:24)
>     at retryTask (webpack-internal:///(sc_server)/./node_modules/next/dist/compiled/react-server-dom-webpack/cjs/react-server-dom-webpack-server.edge.development.js:1882:38)
>     at performWork (webpack-internal:///(sc_server)/./node_modules/next/dist/compiled/react-server-dom-webpack/cjs/react-server-dom-webpack-server.edge.development.js:1920:21)
>     at Timeout.eval [as _onTimeout] (webpack-internal:///(sc_server)/./node_modules/next/dist/compiled/react-server-dom-webpack/cjs/react-server-dom-webpack-server.edge.development.js:1429:28)
>     at listOnTimeout (node:internal/timers:569:17)
>     at process.processTimers (node:internal/timers:512:7) {
>   digest: '135229182'
> }
> null
